src/Purchase.java

```diff
@@ -0,0 +1,61 @@
+public class Purchase  implements Comparable<Purchase>{
+    public final static String NAME = "Java book";
+    public final static int PRICE = 4700;
+    private int amount;
+    private int discount;
+    private WeekDay weekDay;
+
+    public Purchase(){
+    }
+
+    public Purchase(int amount, int discount, WeekDay weekDay){
+        this.amount = amount;
+        this.discount = discount;
+        this.weekDay = weekDay;
+    }
+
+    public Purchase(int amount, int discount, int day){
+        this(amount, discount, WeekDay.values()[day]);
+    }
+
+    public int getAmount() {
+        return amount;
+    }
+
+    public int getDiscount() {
+        return discount;
+    }
+
+    public WeekDay getWeekDay(){
+        return weekDay;
+    }
+
+    public void setAmount(int amount) {
+        this.amount = amount;
+    }
+
+    public void setDiscount(int discount) {
+        this.discount = discount;
+    }
+
+    public void setWeekDay(WeekDay weekDay) {
+        this.weekDay = weekDay;
+    }
+
+    public int getCost(){
+        return Math.round((float)PRICE * amount * (100 - discount) / 100 /100) * 100;

```

Comment:
do not use float at all. Also, you don't need type casting here .The formula is
Math.round(PRICE * amount* (100 - discount) / 100);